### PR TITLE
Add listing and uninstalling DBMS plugins

### PR DESCRIPTION
**Pull request by nglgzz**
### Please check if the PR fulfills these requirements
- [x] The commit message follows our [guidelines](https://github.com/neo4j-devtools/relate/blob/master/CONTRIBUTING.md)
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

### What kind of change does this PR introduce?
Feature

### What is the current behavior?
There is no functionality for uninstalling or listing Neo4j plugins. 

We don't have debug logs anywhere in the application (except for stack traces when an error occurs).

### What is the new behavior?
 Added functionality and e2e tests to both the common and CLI packages to list and uninstall plugins.

A new `DEBUG` event is now available for messages that can be discarded in most cases, but can help when debugging. All commands in the CLI package are now listening to this event and will log messages according to the `DEBUG` environment variable. Examples: 
```bash
# Show no debug logs
$ relate ...

# Show all debug logs
$ DEBUG=* relate ...

# Show only relate logs (effectively filtering out the nestjs ones)
$ DEBUG=relate* relate ...
```

### Does this PR introduce a breaking change?
No

### CODE REVIEW

File: packages/common/src/entities/environments/environment.constants.ts [hunk 1]
```diff
@@ -39,7 +39,8 @@ export const ZULU_JAVA_DOWNLOAD_URL = 'https://cdn.azul.com/zulu/bin/';
 export const NEO4J_BIN_DIR = 'bin';
 export const NEO4J_BIN_FILE = process.platform === 'win32' ? 'neo4j.bat' : 'neo4j';
 export const NEO4J_ADMIN_BIN_FILE = process.platform === 'win32' ? 'neo4j-admin.bat' : 'neo4j-admin';
-export const NEO4J_PLUGIN_SOURCES_URL = 'https://dist.neo4j.org/relate/official-plugin-sources.json';
+export const NEO4J_PLUGIN_SOURCES_URL =
+    'https://s3-eu-west-1.amazonaws.com/dist.neo4j.org/relate/official-plugin-sources.json';
```
Comment (nglgzz) [inline packages/common/src/entities/environments/environment.constants.ts:43]: This URL is pointing to the same resource as the previous one, but this one is updated instantly when the file is updated, while the other one takes some hours/days to update when we upload a new version.

File: packages/cli/src/modules/dbms-plugin/dbms-plugin.e2e.ts [hunk 2]
```diff
@@ -85,4 +88,29 @@ describe('$relate dbms', () => {
...
+        expect(JSON.parse(ctx.stdout).length).toEqual(2);
+        expect(JSON.parse(ctx.stdout)).toContainEqual({
+            name: 'plugin-test',
+            isOfficial: 'false',
+            homepageUrl: 'https://github.com/neo4j-contrib/neo4j-apoc-procedures',
+            version: '4.0.0.17',
+        });
+    });
+
+    test.stdout().it('uninstalls plugin', async (ctx) => {
```
Comment (jk05) [inline packages/cli/src/modules/dbms-plugin/dbms-plugin.e2e.ts:104]: worth adding a test for when no such plugin exists for the dbms when attempting to uninstall?

Comment (nglgzz) [reply packages/cli/src/modules/dbms-plugin/dbms-plugin.e2e.ts:104]: good point, I'll add that

File: packages/cli/src/base.command.ts [hunk 3]
```diff
@@ -28,6 +35,8 @@ export default abstract class BaseCommand extends Command {
         const cliExtensions = loadExtensionsFor(EXTENSION_TYPES.CLI, flags.environment);
         const systemConfig: ISystemModuleConfig = {defaultEnvironmentNameOrId: flags.environment};
 
+        registerHookListener(HOOK_EVENTS.DEBUG, (msg) => this.debug(msg));
```
Comment (jk05) [inline packages/cli/src/base.command.ts:38]: 👍

**Review by jk05 — APPROVED** [review]

**Review by jk05 — APPROVED** [review]

**Comment by nglgzz** [comment]
[[Kick-off] relate handling DBMS plugins](https://trello.com/c/LDjRztj5/1569-kick-off-relate-handling-dbms-plugins)

**Pull request merged**